### DISCUSSION TITLE
Inheritance mapping works inconsistently

### DISCUSSION THREAD

**Discussion by Paulskit** [LuckyPennySoftware/AutoMapper#3776] [NA]
### Version: x.y.z
Automapper 10.1.1

### Expected behavior
Regardless of order, fields should be mapped consistently.

### Actual behavior
If derived class goes first, mapping works correctly for this and all further mappings.
If derived class goes after base class, mapping works incorrectly for this array and any other array afterwards (I assume it just caches the incorrect plan).

### Steps to reproduce
Here is the [Fiddle](https://dotnetfiddle.net/n4YDoD) to reproduce.

Just change `works` to `doesNotWork` on line 27 to reproduce the issue. Notice `AdditionalProperty` value in the output.

**Comment by lbargaoanu** [comment]
Try the [MyGet](https://docs.automapper.org/en/latest/The-MyGet-build.html) build.

**Reply by lbargaoanu** [reply]
Thanks for the repro. I've added it as a test just in case.

**Reply by Paulskit** [reply]
Will try. Thank you!
